pydy/tests/test_joints.py

```diff
@@ -0,0 +1,275 @@
+        self.new_pinjoint = PinJoint('pinjoint', self.parent, self.child)
+        self.new_pinjoint._apply_joint()
+        # part 1
+        assert self.child.parent == self.parent
+        assert self.parent.child == self.child
+
+        # part 2
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
+        assert self.pinjoint.parent_joint_vector == parent_point
+        assert self.pinjoint.child_joint_vector == child_point
+
+        # part 4
+        parent_axis = self.pinjoint.parent_axis
+        child_axis = self.pinjoint.child_axis
+        assert acos(dot(parent_axis, child_axis)/(parent_axis.magnitude() * child_axis.magnitude())) == 0
+
+
+class TestSlidingJoint():
+    def setup(self):
+        self.parent = Body('parent')
+        self.child = Body('child')
+        self.slidingjoint = SlidingJoint('slidingjoint', self.parent, self.child)
+
+    def test_slidingjoint_init(self):
+        # Default values
+        assert self.slidingjoint.parent == self.parent
+        assert self.slidingjoint.child == self.child
+        assert self.slidingjoint.direction1 == self.parent.x
+        assert self.slidingjoint.direction2 == self.child.x
+        assert self.slidingjoint.parent_point == self.parent.masscenter
+        assert self.slidingjoint.child_point == self.child.masscenter
+
+    def test_slidingjoint_paramters(self):
+        # custom parameters
+        self.slidingjoint = SlidingJoint('slidingjoint', self.parent, self.child, par_point_vec_tuple=(1,0,0),
+                                         child_point_vec_tuple=(0,1,0), direction1='x', direction2='y')
+        assert self.pinjoint.direction1 == self.parent.frame.x
+        assert self.pinjoint.direction2 == self.child.frame.y
+        point1 = self.parent.masscenter.locatenew(self.parent.frame, self.parent.frame.x)
+        point2 = self.child.masscenter.locatenew(self.child.frame, self.child.frame.y)
+        assert self.pinjoint.parent_point == point1
+        assert self.pinjoint.child_joint_piont == point2
+        assert self.pinjoint.parent_joint_vector == self.parent.frame.x
+        assert self.pinjoint.child_joint_vector == self.child.frame.y
+
+    def test_slidingjoint_functions(self):
+        self.slidingjoint._set_parent_child_rel()
+        assert self.parent.child == self.child
+        assert self.child.parent == self.parent
+
+        self.slidingjoint._locate_joint_point()
+        child_masscenter = self.child.masscenter
+        parent_masscenter = self.parent.masscenter
+        parent_point = parent_masscenter.locatenew(self.name + '_Point', self.parent_joint_vector)
+        child_point = child_masscenter.locatenew(self.name + '_Point', self.child_joint_vector)
```

Comment:
What is the `child_joint_vector`? The vector the sliding happens along or the vector to the point the vector lies on?